mlir/include/mlir/Interfaces/LoopLikeInterface.td

```diff
@@ -220,6 +220,31 @@ def LoopLikeOpInterface : OpInterface<"LoopLikeOpInterface"> {
       /*defaultImplementation=*/[{
         return ::mlir::failure();
       }]
+    >,
+    InterfaceMethod<[{
+        Add a zero-trip-check around the loop to check if the loop body is ever
+        run and return the new loop inside the check. The loop body is moved
```

Comment:
Why does this method create a new loop op? Could the loop op be moved into the zero-trip-check?

Reply:
Sometimes it is easier to create a new loop because we need to rotate the loop (`while {}` -> `do {} while`); otherwise the condition ops will be run twice and might have unwanted side effect. See the implementation of while loop as a example: #80349

We can still rotate the loop without creating a new op, but that will involve many tricky block arguments/operands updates. So I think it might be easier to allow this method to return a new loop op.

Reply:
This sounds like an implementation detail. Can an implementation choose to move over the loop? I think this would be easier for `scf.for`. In that case I would just write that the returned loop could be the same loop (moved) or a new loop (and the the old loop is erased).

Reply:
SG. Updated the comments